data-prepper-plugins/opensearch-source/src/main/java/org/opensearch/dataprepper/plugins/source/opensearch/OpenSearchSource.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.source.opensearch;
+
+import co.elastic.clients.elasticsearch.ElasticsearchClient;
+import org.opensearch.client.opensearch.OpenSearchClient;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPlugin;
+import org.opensearch.dataprepper.model.annotations.DataPrepperPluginConstructor;
+import org.opensearch.dataprepper.model.buffer.Buffer;
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.model.source.Source;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@DataPrepperPlugin(name="opensearch", pluginType = Source.class , pluginConfigurationType =OpenSearchSourceConfiguration.class )
+public class OpenSearchSource implements Source<Record<Event>> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpenSearchSource.class);
+
+    private static final String OPEN_SEARCH = "opensearch";
+
+    private OpenSearchClient openSearchClient;
+
+    private ElasticsearchClient elasticsearchClient;
+
+    private final OpenSearchSourceConfiguration openSearchSourceConfiguration;
+
+    @DataPrepperPluginConstructor
+    public OpenSearchSource(final OpenSearchSourceConfiguration openSearchSourceConfiguration) {
+        this.openSearchSourceConfiguration = openSearchSourceConfiguration;
+    }
+
+    @Override
+    public void start(final  Buffer<Record<Event>> buffer) {
+        if (buffer == null) {
+            throw new IllegalStateException("Buffer provided is null");
+        }
+        startProcess(openSearchSourceConfiguration,buffer);
+    }
+
+    private void startProcess(final OpenSearchSourceConfiguration openSearchSourceConfiguration,final Buffer<Record<Event>> buffer)  {
```

Comment:
I would recommend pulling out the business logic in this class into a separate service. That way this main class can construct all the required classes and execution can happen separately. This will help with unit testing down the road as well as we can inject mocks for classes which create connections to external resources. Let me know if you have questions on this.